contracts/src/instances/lseth/LsETHBase.sol

```diff
@@ -0,0 +1,120 @@
+
+/// @author DELV
+/// @title LsETHHyperdrive
+/// @notice The base contract for the lsETH Hyperdrive implementation.
+/// @dev River has it's own notion of shares to account for the accrual of
+///      interest on the ether pooled in the River protocol. Instead of
+///      maintaining a balance of shares, this integration can simply use River's
+///      shares directly.
+/// @custom:disclaimer The language used in this code is for coding convenience
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+abstract contract LsETHBase is HyperdriveBase {
+    using FixedPointMath for uint256;
+
+    /// @dev The lsETH contract.
+    IRiverV1 internal immutable _river;
+
+    /// @notice Instantiates the lsETH Hyperdrive base contract.
+    /// @param __river The lsETH contract.
+    constructor(IRiverV1 __river) {
+        _river = __river;
+    }
+
+    /// Yield Source ///
+
+    /// @dev Accepts a deposit from the user in base.
+    /// @return sharesMinted The shares that were minted in the deposit.
+    /// @return refund The amount of ETH to refund. This should be zero for
+    ///         yield sources that don't accept ETH.
+    function _depositWithBase(
+        uint256, // unused
+        bytes calldata // unused
+    ) internal pure override returns (uint256, uint256) {
+        revert IHyperdrive.UnsupportedToken();
+    }
+
+    /// @dev Process a deposit in vault shares.
+    /// @param _shareAmount The vault shares amount to deposit.
+    function _depositWithShares(
+        uint256 _shareAmount,
+        bytes calldata // unused
+    ) internal override {
+        // Transfer lsETH shares into the contract.
+        _river.transferFrom(msg.sender, address(this), _shareAmount);
+    }
+
+    /// @dev Process a withdrawal in base and send the proceeds to the
+    ///      destination.
+    function _withdrawWithBase(
+        uint256, // unused
+        address, // unused
+        bytes calldata // unused
+    ) internal pure override returns (uint256) {
+        // lsETH withdrawals aren't necessarily instantaneous. Users that want
+        // to withdraw can manage their withdrawal separately.
+        revert IHyperdrive.UnsupportedToken();
+    }
+
+    /// @dev Process a withdrawal in vault shares and send the proceeds to the
+    ///      destination.
+    /// @param _shareAmount The amount of vault shares to withdraw.
+    /// @param _destination The destination of the withdrawal.
+    function _withdrawWithShares(
+        uint256 _shareAmount,
+        address _destination,
+        bytes calldata // unused
+    ) internal override {
+        // Transfer the lsETH shares to the destination.
+        _river.transfer(_destination, _shareAmount);
+    }
+
+    /// @dev We override the message value check since this integration is
+    ///      payable.
+    function _checkMessageValue() internal pure override {}
```

Comment:
The integration shouldn't be payable since we don't accept the base option. Let's lock this down.